tests/unittests/test_features.py

```diff
@@ -50,7 +115,11 @@ def test_istft():
 
 def test_filterbank():
 
+    tollerance_th = 5e-05
```

Comment:
there is one l in tolerance.  not sure what th means.

Reply:
This is because I sometimes observe a non-deterministic behavior of `torch.mm` when processing sentences of different lengths (see my message above). 
Basically, when I process the a sentence alone vs the same sentence in a batch, `torch.mm` might provide slightly different results. This doesn't happen every time and might depend on the machine one is using (for instance, I observed this occasionally on Colab and other machines). Not sure to understand why. I think it could be due to the fact that cudnn is using different algorithms based on the characteristics of the matrix (e.g, number of time steps) for improving speed.

You can take a look at the colab below for more info:

https://colab.research.google.com/drive/1lklKrRRYTKTXwMbFMh62biQokjmA95is?usp=sharing

Reply:
> This is because I sometimes observe a non-deterministic behavior of torch.mm when processing sentences of different lengths (see my message above).

You may find 
```
torch.use_deterministic_algorithms(mode)
```
is helpful in your case. You can set `mode=True` to make `torch.mm` deterministic.
See https://pytorch.org/docs/stable/generated/torch.use_deterministic_algorithms.html#torch.use_deterministic_algorithms

Reply:
It looks like this makes the code deterministic only when working with CUDA, but we should run the unittest on cpu (the CI server doesn't have a GPU)